subcommand/consul-sidecar/command_test.go

```diff
@@ -79,6 +79,67 @@ func TestRunSignalHandlingRegistration(t *testing.T) {
 	}
 }
 
+func TestRunSignalHandlingMetricsOnly(t *testing.T) {
+	cases := map[string]os.Signal{
+		"SIGINT":  syscall.SIGINT,
+		"SIGTERM": syscall.SIGTERM,
+	}
+	for name, signal := range cases {
+		t.Run(name, func(t *testing.T) {
+			ui := cli.NewMockUi()
+			cmd := Command{
+				UI: ui,
+			}
+
+			randomPorts := freeport.MustTake(1)
+			// Run async because we need to kill it when the test is over.
+			exitChan := runCommandAsynchronously(&cmd, []string{
+				"-enable-service-registration=false",
+				"-enable-metrics-merging=true",
+				"-merged-metrics-port", fmt.Sprint(randomPorts[0]),
+				"-service-metrics-port", "8080",
+				"-service-metrics-path", "/metrics",
+			})
+
+			// Keep an open connection to the server by continuously sending bytes
+			// on the connection so it will have to be drained.
+			var conn net.Conn
+			var err error
+			retry.Run(t, func(r *retry.R) {
+				conn, err = net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", randomPorts[0]))
+				if err != nil {
+					require.NoError(r, err)
+				}
+			})
+			go func() {
+				for {
+					_, err := conn.Write([]byte("hello"))
+					if err != nil {
+						break
+					}
+				}
+			}()
+
+			// Send a signal to consul-sidecar. The merged metrics server can take
+			// up to metricsServerShutdownTimeout to finish cleaning up.
+			cmd.sendSignal(signal)
+
+			// Will need to wait for slightly longer than the shutdown timeout to
+			// make sure that the command has exited shortly after the timeout.
+			waitForShutdown := metricsServerShutdownTimeout + 100*time.Millisecond
+
+			// Assert that it exits cleanly or timeout.
+			select {
+			case exitCode := <-exitChan:
+				require.Equal(t, 0, exitCode, ui.ErrorWriter.String())
+			case <-time.After(waitForShutdown):
+				// Fail if the signal was not caught.
+				require.Fail(t, "timeout waiting for command to exit")
+			}
+		})
+	}
+}
+
 func TestRunSignalHandlingAllProcessesEnabled(t *testing.T) {
 	cases := map[string]os.Signal{
 		"SIGINT":  syscall.SIGINT,
```

Comment:
Commenting here because I can't comment on the exact line: I think we'd need to add a test to flag validation to check that the service registration flags are validated only if `-enable-service-registration` is set.

Reply:
The flag validation tests are structured to test only error cases (if they don't error the test would run forever). However the signal handling tests implicitly test the end to end behavior of the non-error cases, including when -enable-service-registration is set to false. WDYT?

Reply:
Oh, you're right! Nevermind!